### PR TITLE
Add docs for the rest of the files in smaug/core.

### PR DESCRIPTION
**Pull request by xyzsam**

### CODE REVIEW

File: smaug/core/operator.h [hunk 1]
```diff
@@ -34,18 +37,36 @@ class Operator {
     virtual bool validate() {
         return validateInputsOutputs() && opType != OpType::UnknownOp;
     }
+
+    /**
+     * Creates all the input and output tensors specific to this operator.
+     *
+     * This should only be called once the Operator is fully initialized with
+     * all required parameters. It is responsible for creating only the tensors
+     * it "owns". All tensors own their output tensors, but not necessarily all
```
Comment (yaoyuannnn) [inline smaug/core/operator.h:46]: All **operators** own their output tensors.

File: smaug/core/operator.h [hunk 2]
```diff
@@ -76,14 +97,30 @@ class Operator {
...
     std::string name;
     OpType opType;
+    /** The BGL Vertex corresponding to this Operator. */
     Vertex vertex;
     Workspace* workspace;
+    /** The number of tensors that this operator is waiting on before it can be
+     * scheduled. */
     int numPendingInputs;
-    // Host memory access types for different data.
+    /** The memory interface over which the inputs are expected to arrive. */
```
Comment (yaoyuannnn) [inline smaug/core/operator.h:123]: The memory interface over which the inputs/outputs are transferred.

File: smaug/core/operator.h (hunk 1 above)
Comment (xyzsam) [reply smaug/core/operator.h:46]: Done. Also clarified "own".

File: smaug/core/operator.h (hunk 2 above)
Comment (xyzsam) [reply smaug/core/operator.h:123]: This comment will only show up for the `inputMemType` field in Doxygen. I've copied the doc to the other fields.

**Pull request merged**